### PR TITLE
Authenticated user in form request only

### PR DESCRIPTION
**Pull request by luchaos**
Remove most occurrences where the username is passed in a hidden form field and used in requests. 

Exceptions are forms where it is an actual input field meant to be provided by the acting user.

### CODE REVIEW

File: public/js/all.js [hunk 1]
```diff
@@ -372,22 +372,6 @@ jQuery(document).ready(function onReady($) {
     $('.searchusericon').attr('src', '/UserPic/' + ui.item.id + '.png');
     return false;
   });
-
-  var $resetForm = $('#resetform');
```
Comment (luchaos) [inline public/js/all.js:376]: Given the age of this codebase, still using jquery (for the better), we somehow happen to use javascript in a quite inflationary fashion. Simple HTML forms are much easier to reason about and easier to respond to in the backend. Let's get rid of that if possible. There's no "ajax fanciness" to be gained here.

File: public/managehashes.php [hunk 2]
```diff
@@ -49,7 +50,7 @@ function UpdateHashDetails(user, hash) {
     $warning.html('Status: updating...');
```
Comment (luchaos) [inline public/managehashes.php:50]: All of this might as well be just HTML forms if I'm not mistaken.

File: public/request/game/modify.php [hunk 3]
```diff
@@ -1,48 +1,53 @@
...
+        echo "OK";
         exit;
-    } else {
-        if ($field == 3) { // Only return status when unlinking hash
-            echo "FAILED!";
-            exit;
-        }
-        header("location: " . getenv('APP_URL') . "/game/$gameID?e=errors_in_modify_game");
+    }
+    header("location: " . getenv('APP_URL') . "/game/$gameID?e=modify_game_ok");
```
Comment (luchaos) [inline public/request/game/modify.php:43]: Didn't find enough time to untangle this but this request script should _either_ response to ajax requests (with structured json data if possible, not just echo'd strings) _or_ form requests that issue redirects. I'd prefer the latter.
An attempt for simplification will probably follow in a later PR.

File: public/userInfo.php [hunk 4]
```diff
@@ -371,18 +371,18 @@ function resize() {
 
             $newValue = $userIsUntracked ? 0 : 1;
             echo "<tr><td>";
-            echo "<form method='post' action='/request/user/update.php' enctype='multipart/form-data'>";
```
Comment (luchaos) [inline public/userInfo.php:374]: These forms do not post multipart form-data

File: public/request/game/modify.php (hunk 3 above)
Comment (Jamiras) [reply public/request/game/modify.php:43]: I'm going to be touching this as part of the game audit comments feature. What exactly are you looking for here?

Comment (luchaos) [reply public/request/game/modify.php:43]: https://github.com/RetroAchievements/RAWeb/pull/1017/files#diff-52e0986859da92ea38c57b01ddcee4160047d764052aeb975ffe7b3cc62f0632

Basic forms and location header redirects if possible (_If_ it makes sense to implement it with javascript XHR requests then it should at least respond with JSON. we really shouldn't be using javascript for form requests though imo)

**Review by Jamiras — APPROVED** [review]

**Pull request merged**